apollo-compiler/src/test/kotlin/com/apollostack/compiler/GraphqlCompilerTest.kt

```diff
@@ -11,167 +11,70 @@ class GraphqlCompilerTest {
   private val compiler = GraphqlCompiler()
 
   @Test fun heroName() {
-    compiler.write("src/test/data/HeroName.json")
-    val outputFile = File("build/generated/source/apollo/test/HeroName.java")
-    assertThat(outputFile.readText()).isEqualTo(
-        """package test;
+    val queryFile = File("src/test/graphql/com/apollostack/compiler/query/HeroName.json")
+    val actualFile = File("build/generated/source/apollo/com/apollostack/compiler/query/HeroName.java")
+    val expectedFile = File("src/test/graphql/com/apollostack/compiler/query/HeroNameExpected.java")
 
-import java.lang.String;
+    compiler.write(queryFile)
+    assertThat(actualFile.readText()).isEqualTo(expectedFile.readText())
 
-public interface HeroName {
-  Character hero();
-
-  interface Character {
-    String name();
-  }
-}
-""")
-    val source = JavaFileObjects.forSourceLines("test.HeroName", outputFile.readLines())
+    val source = JavaFileObjects.forSourceLines("com.apollostack.compiler.query.HeroName", actualFile.readLines())
     assertAbout(javaSources())
-        .that(listOf(source))
-        .compilesWithoutError()
+      .that(listOf(source))
+      .compilesWithoutError()
   }
 
   @Test fun twoHeroes() {
-    compiler.write("src/test/data/TwoHeroes.json")
-    val outputFile = File("build/generated/source/apollo/test/TwoHeroes.java")
-    assertThat(outputFile.readText()).isEqualTo(
-"""package test;
-
-import java.lang.String;
-
-public interface TwoHeroes {
-  Character r2();
+    val queryFile = File("src/test/graphql/com/apollostack/compiler/query/TwoHeroes.json")
+    val actualFile = File("build/generated/source/apollo/com/apollostack/compiler/query/TwoHeroes.java")
+    val expectedFile = File("src/test/graphql/com/apollostack/compiler/query/TwoHeroesExpected.java")
 
-  Character luke();
+    compiler.write(queryFile)
+    assertThat(actualFile.readText()).isEqualTo(expectedFile.readText())
 
-  interface Character {
-    String name();
-  }
-}
-""")
-    val source = JavaFileObjects.forSourceLines("test.TwoHeroes", outputFile.readLines())
+    val source = JavaFileObjects.forSourceLines("com.apollostack.compiler.query.TwoHeroes", actualFile.readLines())
     assertAbout(javaSources())
-        .that(listOf(source))
-        .compilesWithoutError()
+      .that(listOf(source))
+      .compilesWithoutError()
   }
 
   @Test fun heroDetails() {
-    val compiler = GraphqlCompiler()
-    compiler.write("src/test/data/HeroDetails.json")
-    val outputFile = File("build/generated/source/apollo/test/HeroDetails.java")
-    assertThat(outputFile.readText()).isEqualTo(
-      """package test;
-
-import java.lang.Integer;
-import java.lang.String;
-import java.util.List;
-
-public interface HeroDetails {
-  Character hero();
+    val queryFile = File("src/test/graphql/com/apollostack/compiler/query/HeroDetails.json")
```

Comment:
nit: I think I'd move the test files to `src/test/graphql/com/example/` instead, so it's clear that those are test files. 
I think using `com/apollostack/compiler` makes a bit confusing since those are not actual real files, just test data